proto/opamp.proto

```diff
@@ -817,3 +833,33 @@ message AgentConfigFile {
     // example "text/yaml".
     string content_type = 2;
 }
+
+/////////////////////////////////////////////////////////////////////////////////////
+// Custom messages
+/////////////////////////////////////////////////////////////////////////////////////
+
+message CustomCapabilities {
+    // A list of custom capabilities that are supported. Each capability is a reverse FQDN
+    // with optional version information that uniquely identifies the custom capability
+    // and should match a capability specified in a supported CustomMessage.
+    // Status: [Development]
+    repeated string capabilities = 1;
+}
+
+message CustomMessage {
+    // A reverse FQDN that uniquely identifies the capability and matches one of the
+    // capabilities in the CustomCapabilities message.
+    // Status: [Development]
+    string capability = 1;
+
+    // Type of message within the capability. The capability defines the types of custom
+    // messages that are used to implement the capability. The type must only be unique
+    // within the capability.
+    // Status: [Development]
+    string type = 2;
```

Comment:
Is the purpose of `type` to aid in decoding `data` field? Otherwise, we could embed the `type` information into the `data`.

Reply:
Yes. It makes handling the messages easier because you can switch on the type and decode into a struct that corresponds to the message type.